src/wallet/walletdb.cpp

```diff
@@ -800,19 +800,31 @@ struct MWCoin {
     {
         READWRITE(coin.commitment);
         READWRITE(coin.amount);
-        READWRITE(coin.key.bip32Path);
-        READWRITE(coin.key.keyBytes);
+
+        bool fKey = coin.key == boost::none;
+        READWRITE(fKey);
+        if (fKey) {
+            if (ser_action.ForRead()) {
+                libmw::PrivateKey key;
+                READWRITE(key.bip32Path);
+                READWRITE(key.keyBytes);
+                coin.key = boost::make_optional(std::move(key));
+            } else {
+                READWRITE(coin.key->bip32Path);
+                READWRITE(coin.key->keyBytes);
+            }
+        }
     }
 };
 
 bool WalletBatch::WriteMWCoin(const libmw::Coin& coin)
 {
-    return WriteIC(std::make_pair(std::string("mweb_tx"), coin.commitment), MWCoin{coin});
+    return WriteIC(std::make_pair(std::string("mweb_coin"), coin.commitment), MWCoin{coin});
```

Comment:
let's use an `enum` or at least a `constant` for `mweb_coin`

Reply:
It actually has to be a string in order to follow the existing standard. I can definitely use a const though. even though the other tables don't. It's a better practice, and I don't mind veering a little bit from the existing code.

Reply:
Cool!